akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -22,6 +23,15 @@ final case class HttpChallenge(scheme: String, realm: String,
   def getParams: util.Map[String, String] = params.asJava
 }
 
+// FIXME: Replace these two constructors with a single constructor and default 'params' Map.empty in Akka 3.0
+object HttpChallenge {
+  def apply(scheme: String, realm: Option[String]): HttpChallenge =
+    HttpChallenge(scheme, realm.orNull, Map.empty[String, String])
+
+  def apply(scheme: String, realm: Option[String], params: Map[String, String]): HttpChallenge =
```

Comment:
Not quite, default parameters are a pain for binary compatibility so we'll want to avoid them even in the future, keeping APIs as is thanks to this PR